server/parameters/ServerSubcommandParser.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.vaticle.typedb.core.server.parameters;
+
+import com.vaticle.typedb.core.common.exception.TypeDBException;
+import com.vaticle.typedb.core.server.parameters.util.HelpEntry;
+import com.vaticle.typedb.core.server.parameters.util.SubcommandParser;
+import com.vaticle.typedb.core.server.parameters.util.Option;
+import com.vaticle.typedb.core.server.parameters.util.OptionParser;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.vaticle.typedb.common.collection.Collections.list;
+import static com.vaticle.typedb.common.collection.Collections.set;
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Server.CLI_OPTION_REQUIRED;
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Server.CLI_OPTION_UNRECOGNISED;
+import static com.vaticle.typedb.core.common.iterator.Iterators.iterate;
+
+public class ServerSubcommandParser {
+
+    private static void validateRequiredArgs(Set<OptionParser> requiredParsers, Set<Option> options) {
```

Comment:
argument naming here hints this should be called `validateRequiredOptions`, and below `validateUnrecognisedOptions`

Reply:
Oops! Fixed in 443186cb5e02eba13fd86437dd8e39d299a5e64c.